src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
```

Comment:
I think it would be better to return a new unique_ptr object, not a reference to the internal one. When I call `release_zone()`, I expect the zone to be gone regardless of whether I decide to move it into another pointer object or not. It shouldn't make a noticeable (or any, with compiler optimizations?) difference to performance.